### PR TITLE
Update vcf-validator to 0.9.7 and add support for ARM on OSX

### PR DESCRIPTION
**Pull request by tcezard**
Update of vcf-validator to v0.9.7
Add binaries for ARM architecture on OS-X 

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/vcf-validator/meta.yaml [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
+    sha256: 49aef4841cd7d0913ba12020465830b834ccfe20f2094222c8c41511f6c801ac                                              # [linux]
+  - url: https://github.com/EBIvariation/vcf-validator/releases/download/v{{ version }}/vcf_assembly_checker_linux        # [linux]
+    sha256: a9a00303c05c1fb5a33a5bd1f5c74fb5d8097f92a1f3ee815e19c137c0e26f13                                              # [linux]
+  - url: https://github.com/EBIvariation/vcf-validator/releases/download/v{{ version }}/vcf_validator_macos_x64           # [osx and x86]
+    sha256: 0b97a34fac6cf23951ffc33cdfb7b0602a89b4d5390564c8d816cb83019a3ffc                                              # [osx and x86]
+  - url: https://github.com/EBIvariation/vcf-validator/releases/download/v{{ version }}/vcf_assembly_checker_macos_x64    # [osx and x86]
+    sha256: 1b5404a4239da9f5e5e39bece975e4eea55d87c486121a076226829b79477bc3                                              # [osx and x86]
+  - url: https://github.com/EBIvariation/vcf-validator/releases/download/v{{ version }}/vcf_validator_macos_arm64         # [osx and arm64]
```
Comment (martin-g) [inline recipes/vcf-validator/meta.yaml:17]: Support for OSX ARM64 is coming but not yet available!
See https://github.com/bioconda/bioconda-recipes/pull/46775

Comment (tcezard) [reply recipes/vcf-validator/meta.yaml:17]: Do you mean that the selector does not work ATM?

Comment (martin-g) [reply recipes/vcf-validator/meta.yaml:17]: The selector works but nothing will be build and uploaded.

**Comment by martin-g** [comment]
@BiocondaBot please fetch artifacts

File: recipes/vcf-validator/meta.yaml (hunk 1 above)
Comment (tcezard) [reply recipes/vcf-validator/meta.yaml:17]: I see. Thank you for explaining.
Should I remove it from the build or leave it even if it does not get pulled ?

Comment (martin-g) [reply recipes/vcf-validator/meta.yaml:17]: As you prefer. It is just ignored at the moment, so it does not cause any harm.

Comment (aliciaaevans) [reply recipes/vcf-validator/meta.yaml:17]: The `osx-arm64` builds have been enabled just now! You'll need to add the additional platform configuration: https://bioconda.github.io/developer/aarch64.html

**Comment by aliciaaevans** [comment]
@BiocondaBot please fetch artifacts

**Comment by aliciaaevans** [comment]
Had to merge in the changes from master to get the osx-arm64 build to run.

**Pull request closed without merging**